### PR TITLE
Make markdown toolbar extensible

### PR DESCRIPTION
**Pull request by davwheat**
Replaces #31 because I broke GitHub.

### CODE REVIEW

**Review by askvortsov1 — APPROVED** [review]
respectable

**Comment by dsevillamartin** [comment]
Is this easily extendible? I thought using override/extend on exported functions was hard (cause don't you have to go through `flarum.core.compat` to be able to modify the method?) 

Just occurred to me.

**Comment by davwheat** [comment]
This should work:

```ts
old = markdownToolbarItems;
markdownToolbarItems = function (){
  const items = old();

  items.add(/* ... */);

  return items;
}
```

**Comment by dsevillamartin** [comment]
@davwheat Um, don't think so? Assigning variables from `import` statements don't change what the import refers to afaik... That'd only work for a global variable or an exported object/class.

**Comment by askvortsov1** [comment]
> This should work:
> 
> ```ts
> old = markdownToolbarItems;
> markdownToolbarItems = function (){
>   const items = old();
> 
>   items.add(/* ... */);
> 
>   return items;
> }
> ```

Ah yeah David is right. Until we have an export registry and can actually replace functions in the registry, this won't work.

For some reason I thought this was being added a a method, must have missed this, sorry

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
see above

**Comment by davwheat** [comment]
Hoping that #34 will be merged first, since this PR will need some changes once that is done.

**Comment by davwheat** [comment]
Ready for review now.

File: js/src/forum/index.js [hunk 1]
```diff
@@ -48,35 +49,57 @@ function makeShortcut(id, key, editorDriver) {
...
-    };
-
-    const makeApplyStyle = (id) => {
-      return () => applyStyle(id, this.attrs.composer.editor);
-    };
+  if (TextEditor.prototype.markdownToolbarItems) {
+    override(TextEditor.prototype, 'markdownToolbarItems', markdownToolbarItems);
+  } else {
+    TextEditor.prototype.markdownToolbarItems = markdownToolbarItems;
+  }
```
Comment (askvortsov1) [inline js/src/forum/index.js:96]: Shouldn't we be able to `override` in either case?

Comment (davwheat) [reply js/src/forum/index.js:96]: When I tried that, it seemed to error at runtime.

Comment (davwheat) [reply js/src/forum/index.js:96]: image
image

This is because it calls `bind` on the original function, which doesn't exist.

Comment (askvortsov1) [reply js/src/forum/index.js:96]: Hmm, we might want to change that in core. But makes sense for now.

**Review by askvortsov1 — APPROVED** [review]

File: js/src/forum/index.js (hunk 1 above)
Comment (davwheat) [reply js/src/forum/index.js:96]: I agree

**Review by SychO9 — APPROVED** [review]

**Pull request merged**